scopesim/effects/ter_curves.py

```diff
@@ -92,13 +87,16 @@ def __init__(self, **kwargs):
         if self.meta["ignore_wings"]:
             data = add_edge_zeros(data, "wavelength")
         if data is not None:
+            # Assert that get_data() did not give us an image.
+            assert isinstance(data, Table), "TER Curves must be tables."
             self.surface.table = data
             self.surface.table.meta.update(self.meta)
 
     # ####### added in new branch
 
     def apply_to(self, obj, **kwargs):
         if isinstance(obj, SourceBase):
+            assert isinstance(obj, Source), "Only Source supported."
```

Comment:
I might be missing something obvious here, but why not just check for `Source` instead of `SourceBase` in the first place?
Same of course for `FovVolumeList` below...

Reply:
I did that at first, but then reverted back to this because I wanted to keep the structure of the `apply_to()` method identical to the `apply_to()` method of other Effect classes. I'd rather update all the `apply_to()` methods (in the way you describe) in one sweep, see my second suggestion in #241.